boa3/model/builtin/interop/contract/contracttype.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+from boa3.model.builtin.method.builtinmethod import IBuiltinMethod
+from boa3.model.expression import IExpression
+from boa3.model.method import Method
+from boa3.model.property import Property
+from boa3.model.type.classtype import ClassType
+from boa3.model.variable import Variable
+from boa3.neo.vm.opcode.Opcode import Opcode
+
+
+class ContractType(ClassType):
+    """
+    A class used to represent Neo Contract class
+    """
+
+    def __init__(self):
+        super().__init__('Contract')
+
+        from boa3.model.type.type import Type
+        self._variables: Dict[str, Variable] = {
+            'script': Variable(Type.bytes),
+            'manifest': Variable(Type.dict)
+        }
+        self._constructor: Method = None
+
+    @property
+    def variables(self) -> Dict[str, Variable]:
+        return self._variables.copy()
+
+    @property
+    def properties(self) -> Dict[str, Property]:
+        return {}
+
+    @property
+    def class_methods(self) -> Dict[str, Method]:
+        return {}
+
+    @property
+    def instance_methods(self) -> Dict[str, Method]:
+        return {}
+
+    def constructor_method(self) -> Method:
+        # was having a problem with recursive import
+        if self._constructor is None:
+            self._constructor: Method = ContractMethod(self)
+        return self._constructor
+
+    @classmethod
+    def build(cls, value: Any = None):
+        if value is None or cls._is_type_of(value):
+            return _Contract
+
+    @classmethod
+    def _is_type_of(cls, value: Any):
+        return isinstance(value, ContractType)
+
+
+_Contract = ContractType()
+
+
+class ContractMethod(IBuiltinMethod):
+
+    def __init__(self, return_type: ContractType):
+        identifier = '-Contract__init__'
+        args: Dict[str, Variable] = {}
+        super().__init__(identifier, args, return_type=return_type)
+
+    def validate_parameters(self, *params: IExpression) -> bool:
+        return len(params) == 0
+
+    @property
+    def opcode(self) -> List[Tuple[Opcode, bytes]]:
+        from boa3.neo.vm.type.Integer import Integer
+        return [
+            Opcode.NEWMAP,
```

Comment:
the type of the values of this list is a Tuple[Opcode, bytes], but you inserted just an Opcode here.
when an Opcode doesn't have arguments, you can add the same way the Opcode.PUSH2 and Opcode.PACK in the lines below